components/constraints.js

```diff
@@ -12,8 +12,11 @@ class Constraints extends Component {
     if (!constraints) return <div />;
     return (
       <ul style={{ listStyle: 'none', paddingLeft: 0 }}>
-        {(constraints.get('default') || constraints.get('default') === 0) &&
-          <li>default value: {constraints.get('default')}</li>
+        {constraints.has('default') &&
+          (constraints.get('type') === 'boolean' ?
+            (constraints.get('default') ? 'default value: true' : 'default value: false') :
```

Comment:
All schema properties are optional.  The empty schema `{}` validates everything.  So for instance, this:

```JSON
{
    "default": false
}
```

Is a schema that accepts any value of any time, with no restriction whatsoever.  But it's default value is boolean `false`.

That's pretty straightforward, I think- sometimes you actually do have unrestricted data, if only rarely.

A weirder but still valid schema is:

```JSON
{
    "type": "integer",
    "default": fase
}
```

In this example, the default does not validate.  Which is weird, and of questionable value.  It gets discussed a lot over on the spec project- `"default"` is a bit of a headache because it's not very well-specified as to how you can and can't use it.  See json-schema-org/json-schema-spec#204 and json-schema-org/json-schema-spec#217 for way more discussion than you probably want to know.

A more realistic schema that you would not handle with the current code is 

```JSON
{
    "type": ["boolean", "null"],
    "default": false
}
```

which declares a 3-state field (nullable boolean).  
Basically, a boolean default value should be presented the same way whether the type is present or not, regardless of the type specified.  It's a bit weird, but that's what it is right now.